### PR TITLE
Add i18n to homepage

### PR DESCRIPTION
**Pull request by giovramirez**
Fixes https://github.com/vendure-ecommerce/storefront-qwik-starter/issues/88

### CODE REVIEW

**Comment by giovramirez** [comment]
After updating the Qwik dependencies to 0.102.0, bug 2 is fixed.

File: src/components/footer/footer.tsx [hunk 1]
```diff
@@ -3,32 +3,34 @@ import { Link } from '@builder.io/qwik-city';
...
-		{ name: 'Blog', href: '#' },
-		{ name: 'Corporate responsibility', href: '#' },
-		{ name: 'Press', href: '#' },
-	],
-};
 export default component$(() => {
 	const collections = useContext(APP_STATE).collections.filter(
 		(item) => item.parent?.name === '__root_collection__' && !!item.featuredAsset
 	);
+
```
Comment (giovramirez) [inline src/components/footer/footer.tsx:10]: It needs to move the `navigation` object to the component context in order to get the locale info properly.

**Comment by giovramirez** [comment]
@gioboa pr is ready to review!

Collections on the homepage are still displaying in English, I think we need to add the proper translations in the Vendure server. Do you have access?

**Comment by gioboa** [comment]
I don't have access to the server but @michaelbromley is the admin 😄

**Comment by michaelbromley** [comment]
Hi @giovramirez!
I just pulled this PR locally to test - it works great!

I would be happy to add translations to the demo server, but we need an automated way to do it. [Here is the source of the demo project](https://github.com/vendure-ecommerce/readonlydemo.vendure.io) - it is dockerized and resets every night, so it doesn't make sense to grant you access to manually add translations.

Instead, we should create a function as part of the [reset/populate script](https://github.com/vendure-ecommerce/readonlydemo.vendure.io/blob/master/src/reset-server.ts) which programmatically adds translations to each Product & Collection. 

Would you be interested in attempting this? I would be happy to advise.

**Comment by giovramirez** [comment]
Hello @michaelbromley,

Yes, let's go for it!

Only I can't see the 'readonlydemo.vendure.io' repo

**Comment by gioboa** [comment]
is it ready ready for review?

**Comment by giovramirez** [comment]
Hi @gioboa, yes it is!

However, there's a dependency on translate Products and Collections in the Vendure server. So I'll create a new issue on https://github.com/vendure-ecommerce/vendure-demo repo to track this work there.

**Comment by gioboa** [comment]
Great 😃 can you solve conflicts with the main branch?

**Comment by giovramirez** [comment]
Done!

However `pnpm run preview` is not working anymore. I am facing this error:

```
/storefront-qwik-starter/src/utils/i18n.ts
  97:5  error  Calling use* methods in non-root component  qwik/use-method-usage
```

@gioboa you think it's better to move [this line](https://github.com/vendure-ecommerce/storefront-qwik-starter/pull/91/files#diff-648766b39e18e48addc4d75d1e8c92235f724796ffb67e8e84d8b08aaf437faaR97) to `root.tsx` in order to avoid the linter issue?

**Comment by gioboa** [comment]
Does that solve the issue?

**Comment by giovramirez** [comment]
Yes, I added a new [commit ](https://github.com/vendure-ecommerce/storefront-qwik-starter/pull/91/commits/4c673a6fa8650fa1ee5096bfa6407fbfea3da566)

Please let me know if this proposal looks good to you!

**Comment by gioboa** [comment]
@giovramirez Can you solve these conflicts pls?

**Comment by michaelbromley** [comment]
Great work @giovramirez, thank you!

**Pull request merged**